source/beamline/BioXAS/BioXASBeamline.cpp

```diff
@@ -88,6 +117,45 @@ AMAction3* BioXASBeamline::createScanInitializationAction(AMGenericStepScanConfi
 		if (BioXASBeamlineSupport::usingScaler(configuration)) {
 			scalerInitialization = new AMListAction3(new AMListActionInfo3("BioXAS scaler initialization", "BioXAS scaler initialization"));
 			scalerInitialization->addSubAction(scaler->createContinuousEnableAction3(false)); // Check that the scaler is in single shot mode and is not acquiring.
+
+			// Determine the longest region time for the scan.
+
+			double maxRegionTime = configuration->scanAxisAt(0)->regionAt(0)->regionTime();
+
+			for (int axisIndex = 0, axisCount = configuration->scanAxes().count(); axisIndex < axisCount; axisIndex++) {
+				for (int regionIndex = 0, regionCount = configuration->scanAxisAt(axisIndex)->regionCount(); regionIndex < regionCount; regionIndex++) {
+
+					AMScanAxisRegion *region = configuration->scanAxisAt(axisIndex)->regionAt(regionIndex);
+					AMScanAxisEXAFSRegion *exafsRegion = qobject_cast<AMScanAxisEXAFSRegion*>(region);
+
+					double regionDwell = 0;
+
+					// If the region is an EXAFS region, the dwell time at a particular point could vary.
+					// In this case, the best way to make sure the largest possible dwell time is taken
+					// into account is to compare against the maximumTime() instead of the regionTime().
+
+					if (exafsRegion)
+						regionDwell = double(exafsRegion->maximumTime());
+					else
+						regionDwell = double(region->regionTime());
+
+					if (maxRegionTime < regionDwell)
+						maxRegionTime = regionDwell;
+				}
+			}
+
+			// Determine whether a dark current measurement is needed, by asking each scaler channel detector
+			// if its dark current value would be valid for the longest region time.
+
+			bool requiresDarkCurrentMeasurement = false;
```

Comment:
Oh wait, yes, I misread what you wrote. I think all that's really necessary for a default XANES scan, for example, is a dark current measurement of 1s. However, a user can still take a dark current measurement for as long as they'd like before a scan.